### ISSUE TITLE
Add attoparsec-aeson-2.1 to older LTS snapshots for compatibility

### ISSUE THREAD

**Issue by IamfromSpace** [commercialhaskell/lts-haskell#365]
Hi all,

(copied from an original issue [here](https://github.com/commercialhaskell/stackage/issues/7071))

In hal we currently have an issue where we can't release a new version that would be compatible across old snapshots while also enabling support for nightly. That is, without doing so with a new breaking change release (from 1.0.x.x to 1.1.x.x), which would cause unnecessary noise for users, and I'd like to avoid ([In discussion here](https://github.com/Nike-Inc/hal/pull/122)).

In advancing to aeson-2.2 which moved previous modules to attoparsec-aeson-2.2, the team also created attoparsec-aeson-2.1 as a compatibility release. This theoretically allows a package (such as hal), to support either aeson < 2.2 or aeson = 2.2 simultaneously. When depending on attoparsec-aeson then, it can use either version, to either expose the moved modules or nothing (since older aeson versions already do).

The aeson maintainers [have asked that stackage curators take this on](https://github.com/haskell/aeson/issues/1048#issuecomment-1655906002) and that I open an issue here.

tl;dr; if attoparsec-aeson is available in older LTS snapshots, hal can release a version bump with no impact to users, which would be compatible across the upcoming and last dozen snapshots.

Please let me know if there's any other info I can provide or a way I can assist in making this happen, thanks!

**Comment by juhp** [comment]
I think the only real way is to add `attoparsec-aeson` to `extra-deps` in stack.yaml for older stackage releases.

Basically even if we want to do this, we basically can't technically, since we don't have any way currently to update older LTS releases sadly.

BUT we should definitely add attoparsec-aeson to Nightly and LTS 21 !! ;o)

**Comment by juhp** [comment]
I am adding it to the next nightly.
Could you open a ticket to add it to lts?
(alternatively you could requisition this  one for that purpose if you wish)

**Comment by IamfromSpace** [comment]
> I am adding it to the next nightly

That's great!  That will unblock `hal`'s (and other packages', I'd imagine) ability to support nightly.

I think using this ticket for accepting `attoparsec-aeson` into LTS would make the most sense.  Looking at the [Stackage maintainers FAQ](https://github.com/commercialhaskell/stackage/blob/master/MAINTAINERS.md#adding-a-package-to-an-lts-snapshot), I believe this is the kind of scenario that justifies the incorporation into LTSes.

I think i mispoke when I said "snapshots;" it of course makes sense that older snapshots are immutable.  I think I mean instead the next release of older LTSes--please let me know if there's a more precise way to express this!  For example, lts-20.26 the most recent lts-20 as of now, I was hoping that lts-20.27 would include `attoparsec-aeson` (as well as earlier major LTSes, preferably).

The advantage being that all of the coming releases of old and new LTS snapshots along with nightly could accept the same code for packages like `hal`, and users never have to deal with a new major version release.  Happy to explain this more, or open to hear if there are other viable alternatives.

Please let me know what I can do to help, or if I should re-title this issue or anything else.  Appreciate the assistance!

**Comment by alaendle** [comment]
As mentioned here https://github.com/commercialhaskell/stackage/blob/master/MAINTAINERS.md#adding-a-package-to-an-lts-snapshot, we only maintain the current major LTS; so `attoparsec-aeson` will appear in the next LTS major release (lts-22.0) (since it was added to the current nightly build) and we will add to the current lts, so it will also appear in the next LTS minor release (lts-21.6).

**Comment by alaendle** [comment]
https://www.stackage.org/lts-21.6/package/attoparsec-aeson-2.1.0.0

**Comment by IamfromSpace** [comment]
Ah, I think for some reason I thought that older LTS minor versions continued to be published after release of the next, but reviewing release dates, that's clearly not the case!

Given that, this addresses all my concerns, and really appreciate the addition!
